pkg/kubelet/winstats/perfcounter_nodestats.go

```diff
@@ -110,6 +116,23 @@ func (p *perfCounterNodeStatsClient) startMonitoring() error {
 		p.collectMetricsData(cpuCounter, memWorkingSetCounter, memCommittedBytesCounter, networkAdapterCounter)
 	}, perfCounterUpdatePeriod)
 
+	// Cache the CPU usage every 15 seconds
+	go wait.Forever(func() {
+		cpuCores := runtime.NumCPU()
+		cpuValue, err := cpuCounter.getData()
+		if err != nil {
+			klog.Errorf("Unable to get cpu perf counter data; err: %v", err)
+			return
+		}
+		newValue := p.nodeMetrics.cpuUsageCoreNanoSeconds + uint64((float64(cpuValue)/100.0)*float64(cpuCores)*1e9)
```

Comment:
Could we also avoid converting back and forth between float64 and uint64 if we multiplied by 10 million at the end instead of 1e9 (e.g. combine converting from nano and dividing percent by 100)?

Reply:
Can you also add a unit test for the math (maybe factor just the math out to a helper to make that easier).

Reply:
This calculation is actually the same as the [convertCPUValue](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/winstats/perfcounter_nodestats.go#L182), I changed to call it directly, and probably leave the calculation as it is for now?

And I just added the unit test for the calculation.

Reply:
Ah, I see it's supposed to be an accumulator across the lifetime of the node. Ok then.

Reply:
Thanks for adding a test.

Reply:
@mtaufen I think you are correct here.  `cpuUsageCoreNanoSeconds ` is the cumulative usage over time.  UsageNanoCores, which is an average over the collection period, should be uncs(t2) - uncs(t1) / (t2-t1).  It looks like we are not dividing by the time between collections.

Reply:
Thanks for the explanation! I was misunderstanding the meaning of `UsageNanoCores` and didn't use the average value. I modified the calculation, please take another look.

Reply:
It looks correct to me now.